source/kraken/metrics.cpp

```diff
@@ -151,6 +151,48 @@ Metrics::Metrics(const boost::optional<std::string>& endpoint, const std::string
                                              .Register(*registry)
                                              .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->retrieve_rt_message_duration_histogram = &prometheus::BuildHistogram()
+                                                        .Name("kraken_retrieve_rt_message_duration_seconds")
+                                                        .Help("duration of RT messages retrieval from RabbitMQ")
+                                                        .Labels({{"coverage", coverage}})
+                                                        .Register(*registry)
+                                                        .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->retrieved_rt_message_number_histogram = &prometheus::BuildHistogram()
+                                                       .Name("kraken_retrieve_rt_message_number")
+                                                       .Help("number of RT messages retrieved from RabbitMQ")
+                                                       .Labels({{"coverage", coverage}})
+                                                       .Register(*registry)
+                                                       .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->applied_rt_entity_number_histogram = &prometheus::BuildHistogram()
+                                                    .Name("kraken_applied_rt_entity_number")
+                                                    .Help("number of applied RT entity from a message batch")
+                                                    .Labels({{"coverage", coverage}})
+                                                    .Register(*registry)
+                                                    .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->rt_message_age_min_histogram = &prometheus::BuildHistogram()
+                                              .Name("kraken_rt_message_age_min_seconds")
+                                              .Help("Minimum age of RT message from a batch")
+                                              .Labels({{"coverage", coverage}})
+                                              .Register(*registry)
+                                              .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->rt_message_age_average_histogram = &prometheus::BuildHistogram()
+                                                  .Name("kraken_rt_message_age_average_seconds")
+                                                  .Help("Average age of RT message from a batch")
+                                                  .Labels({{"coverage", coverage}})
+                                                  .Register(*registry)
+                                                  .Add({}, create_exponential_buckets(0.5, 2, 10));
+
+    this->rt_message_age_max_histogram = &prometheus::BuildHistogram()
```

Comment:
As for the message number (now count), the `count` from this histogram measures the number of messages retrieved that have a date, not all of them (not sure the ones from chaos have some), and not all the entities applied.
So it can be interesting to have it, but it's not what I aimed for when I listed the needs.

Reply:
my question is why do you need to know min/max precisely?

Reply:
@xlqian : This metric gives the latency between the reception of RT info from provider to the moment it is available for the travelers (kirin's time is monitored too, to complete it).
So it is quite important for some clients to track it, and `max` or `average` is important and tracked to some clients.
`min` is less important, but it's interesting because it measures minimal latency unrelated to RT-messages themselves.

This is actually the end-goal of improving the RT handling time, the metric we are going to track.

To work on it, especially on max, steps of 128s (~2 min), 256s (~4 min) or "above" is too large (as we want to be able to show a precise max to client - and track it for our work).

Reply:
Something like : 
```
 this->rt_message_average_histogram = &prometheus::BuildHistogram()
                                                  .Name("kraken_rt_message_age_average_seconds")
                                                  .Help(" Age of RT messages")
                                                  .Labels({{"coverage", coverage}})
                                                  .Register(*registry)
                                                  .Add({}, 
prometheus::Histogram::BucketBoundaries{0.1, 0.5, 1,   2,   5,   10, 30, 60, 90, 120, 180});
```

Reply:
I tried adding a unique/more precise histogram (see https://github.com/hove-io/navitia/commit/fd891b3b481f2ad4de3c9571c4866328dca523f2 on branch `metrics_only_one_histogram_rt_message_age`) and the result in NewRelic on max is not good on one test.
As discussed, we will probably move on as-is and decide to dig after the release.